src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -0,0 +1,130 @@
+
+@Service
+public class DeleteCompleteFilesTask {
+    private static final Logger log = LoggerFactory.getLogger(DeleteCompleteFilesTask.class);
+
+    private static final String TASK_NAME = "delete-complete-files";
+
+    private final BlobManager blobManager;
+    private final EnvelopeRepository envelopeRepository;
+    private final EnvelopeProcessor envelopeProcessor;
+
+    public DeleteCompleteFilesTask(
+        BlobManager blobManager,
+        EnvelopeRepository envelopeRepository,
+        EnvelopeProcessor envelopeProcessor
+    ) {
+        this.blobManager = blobManager;
+        this.envelopeRepository = envelopeRepository;
+        this.envelopeProcessor = envelopeProcessor;
+    }
+
+    public void run() {
+        log.info("Started {} task", TASK_NAME);
+
+        for (CloudBlobContainer container : blobManager.listInputContainers()) {
+            try {
+                processCompleteFiles(container);
+            } catch (Exception ex) {
+                log.error(
+                    "Failed to process files from container {}",
+                    container.getName(),
+                    ex
+                );
+            }
+        }
+
+        log.info("Finished {} task", TASK_NAME);
+    }
+
+    private void processCompleteFiles(CloudBlobContainer container) {
+        log.info("Started deleting complete files in container {}", container.getName());
+
+        List<Envelope> envelopes = envelopeRepository.findByContainerAndStatusAndZipDeleted(
+            container.getName(),
+            COMPLETED,
+            false
+        );
+        int successCount = 0;
+        int failureCount = 0;
+        for (Envelope envelope: envelopes) {
+            if (tryProcessCompleteEnvelope(container, envelope)) {
+                successCount++;
+            } else {
+                failureCount++;
+            }
+        }
+
+        log.info(
+            "Finished deleting complete files in container {}, deleted {} files, failed to delete {} files",
+            container.getName(),
+            successCount,
+            failureCount
+        );
+    }
+
+    private boolean tryProcessCompleteEnvelope(CloudBlobContainer container, Envelope envelope) {
+        boolean deleted = false;
+
+        try {
+            CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(envelope.getZipFileName());
+
+            log.info(
+                "File {} (container {}) is complete - deleting",
+                envelope.getZipFileName(),
+                container.getName()
+            );
+
+            if (cloudBlockBlob.exists()) {
+                deleted = cloudBlockBlob.deleteIfExists();
+                if (deleted) {
+                    log.info(
+                        "Deleted file {} from container {}",
+                        envelope.getZipFileName(),
+                        container.getName()
+                    );
+                }
```

Comment:
what if it's `false`? should something be logged about the failure?

Reply:
I think it happens when the file didn't exist - we should be clear about that in logs

Reply:
Plus we can simplify all these conditions if it's true

Reply:
Returns:
true if the blob was deleted; otherwise, false.

Reply:
I looked through existing code (in BlobProcessorTask) and how it was committed. These checks were committed by Luigi and there was some bug fix mentioned in comments. I should be a bit cautious trying to simplify this piece of code.